packages/databyss-ui/modules/SourcesValueList/SourcesValueList.js

```diff
@@ -0,0 +1,101 @@
+import React, { useState } from 'react'
+import ValueListProvider, {
+  ValueListItem,
+} from '@databyss-org/ui/components/ValueList/ValueListProvider'
+import { withSource } from '@databyss-org/services/sources/SourceProvider'
+import { View, Grid, TextControl, List } from '@databyss-org/ui/primitives'
+import _ from 'lodash'
+
+const ControlList = ({ children, ...others }) => (
+  <List horizontalItemPadding="small" {...others}>
+    {children}
+  </List>
+)
+
+const SourcesValueList = ({ source, onValueChange }) => {
+  const [values, setValues] = useState(source)
```

Comment:
Too many intermediate states here. This change event should bubble up.

Reply:
Since `SourceValueList` is a consumer of `withSource` this requires the parent component to use `setSource` in order to update the source, this causes a re-render which looses focus on every key change.